strawberry/mutation.py

```diff
@@ -1,8 +1,123 @@
+from __future__ import annotations
+
 from functools import partial
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    TypeVar,
+)
+from typing_extensions import Annotated, get_args, get_origin
+
+import strawberry
+from strawberry.annotation import StrawberryAnnotation
+from strawberry.arguments import StrawberryArgument
+from strawberry.extensions.field_extension import (
+    AsyncExtensionResolver,
+    FieldExtension,
+    SyncExtensionResolver,
+)
+from strawberry.utils.str_converters import to_camel_case
+
+from .field import StrawberryField, field
+
+if TYPE_CHECKING:
+    from strawberry.types.info import Info
+
+_T = TypeVar("_T")
+
+
+class InputMutationExtension(FieldExtension):
```

Comment:
if we keep only this extension and not the decorator then I think we can move this do another module, maybe `strawberry.utils.mutation` or `strawberry.field_extensions.input_mutation`?

Reply:
Moved it to `strawberry.field_extensions.input_mutation` :)